services/inventory/agents.go

```diff
@@ -325,6 +325,17 @@ func (as *AgentsService) AddMySQLdExporter(ctx context.Context, req *inventorypb
 		if err != nil {
 			return err
 		}
+		if !req.SkipConnectionCheck {
+			service, err := models.FindServiceByID(tx.Querier, req.ServiceId)
+			if err != nil {
+				return err
+			}
+
+			err = as.r.CheckConnectionToService(ctx, service, row)
+			if err != nil {
```

Comment:
```go
if err = as.r.CheckConnectionToService(ctx, service, row); err != nil {
```

I'm not a fan of that syntax myself, but it is idiomatic, especially when only a single value (error) is returned.